### PR TITLE
tmpmail: init at master

### PR DESCRIPTION
**Pull request by legendofmiracles**
###### Motivation for this change
Adds `tmpmail` to nixpkgs

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/tmpmail/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    repo = "tmpmail";
+    rev = "150b32083d36006cf7f496e112715ae12ee87727";
+    sha256 = "sha256-yQ9/UUxBTEXK5z3f+tvVRUzIGrAnrqurQ0x56Ad7RKE=";
+  };
+
+  dontConfigure = true;
+
+  dontBuild = true;
+
+  nativeBuildInputs = [ w3m curl jq makeWrapper ];
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/tmpmail/default.nix:18]:
```suggestion
  nativeBuildInputs = [ makeWrapper ];
```

File: pkgs/applications/networking/tmpmail/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/man
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/tmpmail/default.nix:22]:
```suggestion
    mkdir -p $out/bin $out/man
```

File: pkgs/applications/networking/tmpmail/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    install -Dm755 -t $out/bin tmpmail
+    install -Dm644 tmpmail.1 "$out/share/man/man1/tmpmail.1"
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/tmpmail/default.nix:24]: Please use installManPage.

File: pkgs/applications/networking/tmpmail/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  '';
+
+   meta = with lib; {
+    homepage = "https://github.com/sdushantha/tmpmail";
+    description = "A temporary email right from your terminal written in POSIX sh ";
+    license = licenses.mit;
+    maintainers = [ maintainers.legendofmiracles ];
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/tmpmail/default.nix:35]:
```suggestion
  };
}
```

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119653` at 46ec3c58 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>tmpmail</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `stdenv`.
Near pkgs/applications/networking/tmpmail/default.nix:1:8:
```
  |
1 | { lib, stdenv, fetchFromGitHub, stdenvNoCC, w3m, curl, jq, makeWrapper}:
  |        ^
```
</li>
  </ul>
</details>

**Comment by legendofmiracles** [comment]
Done @SuperSandro2000

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -8779,6 +8779,8 @@ in
 
   tmpwatch = callPackage ../tools/misc/tmpwatch  { };
 
+  tmpmail = callPackage ../applications/networking/tmpmail {  };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:8782]:
```suggestion
  tmpmail = callPackage ../applications/networking/tmpmail { };
```

**Review by SuperSandro2000 — APPROVED** [review]
LGTM when the package related commits are squashed together.

**Comment by legendofmiracles** [comment]
I hope I squashed all commits correctly… 
But it looks good to me now as well @SuperSandro2000

**Pull request merged**